src/Model/API/BEditaClient.php

```diff
@@ -187,11 +187,11 @@ public function authenticate($username, $password)
      * Send a GET request a list of resources or objects or a single resource or object
      *
      * @param string $path Endpoint URL path to invoke
-     * @param array $query Optional query string
-     * @param array $headers Headers
-     * @return array Response in array format
+     * @param array|null $query Optional query string
+     * @param array|null $headers Headers
+     * @return array|null Response in array format
      */
-    public function get($path, $query = [], $headers = [])
+    public function get(string $path, array $query = null, array $headers = null) : ?array
```

Comment:
Shouldn't it be the following?
```
public function get(string $path, ?array $query = null, ?array $headers = null) : ?array
```

Reply:
This applies to both this and the following comments.

You're right, yet I think this works anyway. Consider the following example:

```php
<?php

function test1(array $foo = null)
{
    var_dump($foo);
}

function test2(array $foo)
{
    var_dump($foo);
}

function test3(?array $foo = null)
{
    var_dump($foo);
}

function test4(?array $foo)
{
    var_dump($foo);
}

foreach (['test1', 'test2', 'test3', 'test4'] as $fun) {
    echo "$fun(null)" . PHP_EOL;
    try {
        $fun(null);
    } catch (\Throwable $e) {
        echo sprintf('ERROR [%s]: %s', get_class($e), $e->getMessage()) . PHP_EOL;
    }
}

/* OUTPUT:

test1(null)
NULL
test2(null)
ERROR [TypeError]: Argument 1 passed to test2() must be of the type array, null given, called in <thisfile> on line 26
test3(null)
NULL
test4(null)
NULL

*/
```

The only error is triggered by `test2(null)`, because its first argument is of type `array` and its default value isn't `null`. `test1(null)` works because its first argument is of type is `array`, but the supplied default value is `null`, thus `null` is an accepted value. `test3` and `test4` accept an argument of type `?array`, thus `null` is acceptable whether it's the default value or not.

So, as far as I can tell `?array $myArr = null` and `array $myArr = null` are completely interchangeable. I find the first to be more correct, and the latter to be less verbose. I'm changing it as you've rightfully suggested, but we should keep this convention in mind in the future, since both will work.

Reply:
Well, if both forms are accepted and equivalent, we shouldn't be worried about shorter forms in code, when we eventually forget to use the "explicit/redundant" version `?array $myArr = null` ^_^